docs/API.md

```diff
@@ -456,6 +456,8 @@ Options can be configured as a parameter to the [init()](./interfaces/tracer.htm
 | appsec.wafTimeout           | `DD_APPSEC_WAF_TIMEOUT`                        | `5000`  | Limits the WAF synchronous execution time (in microseconds). |
 | appsec.obfuscatorKeyRegex   | `DD_APPSEC_OBFUSCATION_PARAMETER_KEY_REGEXP`   | -       | A regex to redact sensitive data by its key in attack reports. |
 | appsec.obfuscatorValueRegex | `DD_APPSEC_OBFUSCATION_PARAMETER_VALUE_REGEXP` | -       | A regex to redact sensitive data by its value in attack reports. |
+| telemetryEnabled | `DD_TRACE_TELEMETRY_ENABLED` | -       | Whether to enable capturing telemetry. |
```

Comment:
Why add a public option? This should always be enabled unless we can't enable it because of an issue with a specific environment.

Reply:
The option is still public. Let's maybe skip the config completely and use something internal to pass the option, although I'm not sure this is currently possible so it might require adding an extra layer of config.

Reply:
reworked the logic. Let me know what you think

Reply:
@rochdev did you have time to look at this?